asapdiscovery-ml/asapdiscovery/ml/cli.py

```diff
@@ -0,0 +1,2156 @@
+def build_gat(
+    output_dir: Path | None = None,
+    trainer_config_cache: Path | None = None,
+    optimizer_type: OptimizerType | None = None,
+    lr: float | None = None,
+    weight_decay: float | None = None,
+    momentum: float | None = None,
+    dampening: float | None = None,
+    b1: float | None = None,
+    b2: float | None = None,
+    eps: float | None = None,
+    rho: float | None = None,
+    optimizer_config_cache: Path | None = None,
+    use_wandb: bool | None = None,
+    sweep: bool | None = None,
+    wandb_project: str | None = None,
+    wandb_name: str | None = None,
+    extra_config: list[str] | None = None,
+    grouped: bool | None = None,
+    strategy: StrategyConfig | None = None,
+    pred_readout: ReadoutConfig | None = None,
+    combination: CombinationConfig | None = None,
+    comb_readout: ReadoutConfig | None = None,
+    max_comb_neg: bool | None = None,
+    max_comb_scale: float | None = None,
+    pred_substrate: float | None = None,
+    pred_km: float | None = None,
+    comb_substrate: float | None = None,
+    comb_km: float | None = None,
+    model_config_cache: Path | None = None,
+    in_feats: int | None = None,
+    num_layers: int | None = None,
+    hidden_feats: str | None = None,
+    num_heads: str | None = None,
+    feat_drops: str | None = None,
+    attn_drops: str | None = None,
+    alphas: str | None = None,
+    residuals: str | None = None,
+    agg_modes: str | None = None,
+    biases: str | None = None,
+    allow_zero_in_degree: bool | None = None,
+    es_type: EarlyStoppingType | None = None,
+    es_patience: int | None = None,
+    es_n_check: int | None = None,
+    es_divergence: float | None = None,
+    es_config_cache: Path | None = None,
+    exp_file: Path | None = None,
+    ds_cache: Path | None = None,
+    ds_config_cache: Path | None = None,
+    ds_split_type: DatasetSplitterType | None = None,
+    train_frac: float | None = None,
+    val_frac: float | None = None,
+    test_frac: float | None = None,
+    enforce_one: bool | None = None,
+    rand_seed: int | None = None,
+    ds_split_config_cache: Path | None = None,
+    loss_type: LossFunctionType | None = None,
+    semiquant_fill: float | None = None,
+    loss_config_cache: Path | None = None,
+    auto_init: bool | None = None,
+    start_epoch: int | None = None,
+    n_epochs: int | None = None,
+    batch_size: int | None = None,
+    target_prop: str | None = None,
+    cont: bool | None = None,
+    loss_dict: dict | None = None,
+    device: torch.device | None = None,
+    overwrite_trainer_config_cache: bool = False,
+    overwrite_optimizer_config_cache: bool = False,
+    overwrite_model_config_cache: bool = False,
+    overwrite_es_config_cache: bool = False,
+    overwrite_ds_config_cache: bool = False,
+    overwrite_ds_cache: bool = False,
+    overwrite_ds_split_config_cache: bool = False,
+    overwrite_loss_config_cache: bool = False,
+):
+    _ = build_trainer_gat(
```

Comment:
avoid double passing params if possible and tuck detail into schema, again if possible, but fine if too complex

Reply:
I think this should be done reasonably well now